File: app/src/main/java/com/example/android/rtdp/implementation/java/di/ActivityModule.java
package com.example.android.rtdp.implementation.java.di;

import com.example.android.rtdp.implementation.java.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
@SuppressWarnings("unused")
public abstract class ActivityModule {
    @ContributesAndroidInjector
    abstract MainActivity contributeMainActivity();
}

File: app/src/main/java/com/example/android/rtdp/implementation/kotlin/di/ActivityModule.kt
package com.example.android.rtdp.implementation.kotlin.di

import com.example.android.rtdp.implementation.kotlin.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
@Suppress("unused")
abstract class ActivityModule {
    @ContributesAndroidInjector
    abstract fun contributeMainActivity(): MainActivity
}